pkg/apis/crd/v1alpha1/types.go

```diff
@@ -353,10 +353,15 @@ type NetworkPolicyStatus struct {
 type Rule struct {
 	// Action specifies the action to be applied on the rule.
 	Action *RuleAction `json:"action"`
-	// Set of port and protocol allowed/denied by the rule. If this field is unset
-	// or empty, this rule matches all ports.
+	// Set of port and protocol allowed/denied by the rule. If this field and Protocols
+	// are unset or empty, this rule matches all ports.
 	// +optional
 	Ports []NetworkPolicyPort `json:"ports,omitempty"`
+	// Set of protocol with its specific spec allowed/denied by the rule. If this field
```

Comment:
protocols

I feel we can remove "with its specific spec"

Reply:
Sure. Done.